ivar.c

```diff
@@ -149,8 +149,13 @@ PRIVATE void objc_compute_ivar_offsets(Class class)
 void object_setIvar(id object, Ivar ivar, id value)
 {
 	char *addr = (char*)object;
-	addr += ivar_getOffset(ivar);
-	*(id*)addr = value;
+	unsigned offset;
+	
+	if (ivar && (offset = ivar_getOffset(ivar)))
```

Comment:
libobjc2 coding style follows the Étoilé guidelines and recommends avoiding implicit conversion from pointer to boolean, so please write this as `(ivar != NULL) && ...`.